presto-main/src/main/java/io/prestosql/sql/planner/EffectivePredicateExtractor.java

```diff
@@ -15,12 +15,15 @@
 
 import com.google.common.collect.ImmutableBiMap;
```

Comment:
Can we do something about excessive filter:
```
- Filter[filterPredicate = ("field" IN ('F', 'O'))]
```
I saw this problem before with table scans and partition columns.
When partition values from one side of join were pushed to the other side, then there is always filter above table scan with `col IN partition_values`.
I tried to tackle it in https://github.com/starburstdata/facebook-presto/commit/7081f3eb2943e9d89b27f14bcf27e8610801e31c
but it didn't work unfortunately because inference cannot easily reduce
```
inheritedValue = 1 OR inheritedValue = 2
inheritedValue = tableScanValue
tableScanValue = 1 OR tableScanValue = 2
``` 
to simply `true` (after simplification) and `TupleDomain` doesn't have true `contains` method

Reply:
Unfortunately, there's not much we can do about it, and I haven't investigated it further. On the other hand, it's a filter over a ValuesNode, so it's inexpensive.

Reply:
It could be meaningful though if filter contains 100s of partition values.